ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerTable.java

```diff
@@ -182,6 +182,24 @@ private void buildBloomFilter() {
     this.idBloom = bf;
   }
 
+  /**
+   * Returns the <code>limit</code> peers (at most) bonded closest to the provided target, based on
+   * the XOR distance between the keccak-256 hash of the ID and the keccak-256 hash of the target.
+   *
+   * @param target The target node ID.
+   * @param limit The amount of results to return.
+   * @return The <code>limit</code> closest peers, at most.
+   */
+  public List<DiscoveryPeer> nearestBondedPeers(final Bytes target, final int limit) {
```

Comment:
I presume this is necessary in order to not give remote clients a list of unverified peers from DNSDaemon lookups?

Reply:
This is using the definition of distance from the devP2P spec to find the _limit_ number of closest peers to the _target_.